source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -1580,11 +1816,90 @@ let apply_missing_flows configuration = function
 let from_taint_model_paths taint_model_paths =
   let file_paths =
     PyrePath.get_matching_files_recursively ~suffix:".config" ~paths:taint_model_paths
   in
+  let rec to_yojson_data_model = function
+    | `Null -> `Null
+    | `Bool b -> `Bool b
+    | `String s -> `String s
+    | `Float f -> `Int (int_of_float f)
+    | `A a -> `List (List.map ~f:to_yojson_data_model a)
+    | `O o ->
+        let obj = List.map ~f:(fun (key, value) -> key, to_yojson_data_model value) o in
+        `Assoc obj
+  in
+  let object_with_position_information value decoder =
+    let (line, column), _ = Jsonm.decoded_range decoder in
+    `O
+      [
+        InternalJsonKeys.line_number, `String (string_of_int line);
+        InternalJsonKeys.column_number, `String (string_of_int column);
+        InternalJsonKeys.value, value;
+      ]
+  in
+  let parse_with_jsonm parasable_string =
+    let decode decoder =
+      match Jsonm.decode decoder with
+      | `Lexeme lexeme -> lexeme
+      | `Error error ->
+          raise
+            (JsonmException
+               {
+                 message = Format.asprintf "%a" Jsonm.pp_error error;
+                 position = Jsonm.decoded_range decoder;
+               })
+      | `End
+      | `Await ->
+          raise
+            (JsonmException
+               {
+                 message = "Encountered end of file or input";
+                 position = Jsonm.decoded_range decoder;
+               })
+    in
+    let rec value element key decoder =
+      match element with
+      | `Os -> obj [] key decoder
+      | `As -> arr [] key decoder
+      | (`Null | `Bool _ | `String _ | `Float _) as data_value ->
+          key (object_with_position_information data_value decoder) decoder
+      | _ ->
+          raise
+            (JsonmException
+               {
+                 message = "Encountered unexpected token or element";
+                 position = Jsonm.decoded_range decoder;
+               })
+    and arr values key decoder =
+      match decode decoder with
+      | `Ae -> key (`A (List.rev values)) decoder
+      | element -> value element (fun value -> arr (value :: values) key) decoder
+    and obj ms key decoder =
+      match decode decoder with
+      | `Oe -> key (`O (List.rev ms)) decoder
+      | `Name name -> value (decode decoder) (fun value -> obj ((name, value) :: ms) key) decoder
+      | _ ->
+          raise
+            (JsonmException
+               {
+                 message = "Encountered unexpected token or object sequence";
+                 position = Jsonm.decoded_range decoder;
+               })
+    in
+    let decoder = Jsonm.decoder (`String parasable_string) in
+    let jsonm_results = value (decode decoder) (fun value _ -> value) decoder in
+    (* Convert Jsonm data model to that of YoJSON to make use of the later's utility functions. *)
+    to_yojson_data_model jsonm_results
+  in
```

Comment:
If I understand correctly, we keep using the `Yojson` representation but add some positions in `Assoc` objects?
I understand the logic of doing it this way since that would lead to the least amount of changes but this is pretty ugly and hard to maintain.

Given that the json representation is not too large, I think I would prefer if we declared our own AST with positions, and parse into it.

That could look like (inspired from pyre's old AST):
```ocaml
module Node = struct
  type 'a t = { 
    location: Location.t; (* something to represent locations *)
    value: 'a;
  }
}


module Json = struct
  type expression =
   | Null
   | Bool of bool
   | String of string
   | Float of float
   | List of t list
   | Assoc of (string * t) list

  and type t = expression Node.t
end
```

Reply:
I will experiment this to see if it would work with YoJSON's utility functions.

Reply:
You will have to reimplement those, but it shouldn't be too hard. I would basically dump Yojson entirely and implement a proper library with an AST with locations.

Reply:
Ah, it's gonna be fun :)